stackslib/src/chainstate/stacks/boot/mod.rs

```diff
@@ -166,10 +169,34 @@ pub struct PoxStartCycleInfo {
     pub missed_reward_slots: Vec<(PrincipalData, u128)>,
 }
 
+fn addr_serialize<S: serde::Serializer>(addr: &StacksAddress, s: S) -> Result<S::Ok, S::Error> {
+    s.serialize_str(&addr.to_string())
+}
+
+fn addr_deserialize<'de, D: serde::Deserializer<'de>>(d: D) -> Result<StacksAddress, D::Error> {
+    let addr_str = String::deserialize(d)?;
+    StacksAddress::from_string(&addr_str)
+        .ok_or_else(|| serde::de::Error::custom("Address must be a C32 encoded StacksAddress"))
+}
+
+#[derive(Debug, PartialEq, Clone, Serialize, Deserialize)]
+pub struct NakamotoSignerEntry {
+    #[serde(
+        serialize_with = "addr_serialize",
+        deserialize_with = "addr_deserialize"
+    )]
+    pub signing_address: StacksAddress,
```

Comment:
Per @jferrant, it turns out that this actually needs to be a public key -- DKG needs the public keys before it can begin.